Controller/Adminhtml/System/Config/Save.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Controller\Adminhtml\System\Config;
+
+class Save extends \Magento\Config\Controller\Adminhtml\System\Config\Save
+{
+    /**
+     * Get groups for save
+     *
+     * @return array|null
+     */
+    protected function _getGroupsForSave()
+    {
+        $groups = parent::_getGroupsForSave();
+
+        return $this->handleDeactivatedSerializedArrays($groups);
+    }
+
+    private function handleDeactivatedSerializedArrays($groups)
+    {
+        if (isset($groups['instant']['fields']['is_instant_enabled']['value'])
+                && $groups['instant']['fields']['is_instant_enabled']['value'] == '0') {
+            if (isset($groups['instant']['fields']['facets'])) {
```

Comment:
How about specifying names of the fields in a property, eg. `$this->instantSerializedValues = ['facets', 'sorts'];` and then iterate over it and don't write a specific condition for each field. It'll be better to maintain and the code will be shorter and more readable.

The same could apply to autocomplete fields.